### PR TITLE
[WIP] Document KubeletEnsureSecretPulledImages feature

### PR DESCRIPTION
**Pull request by pacoxu**
for https://github.com/kubernetes/enhancements/issues/2535

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -129,6 +129,7 @@ For a reference to old feature gates that are removed, please refer to
 | `KMSv2KDF` | `false` | Beta | 1.28 | |
 | `KubeProxyDrainingTerminatingNodes` | `false` | Alpha | 1.28 | |
 | `KubeletCgroupDriverFromCRI` | `false` | Alpha | 1.28 | |
+| `KubeletEnsureSecretPulledImages` | `false` | Alpha | 1.29 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:132]: Each and every gate needs a description that can be found at the end of this file, in alphabeta order.

**Comment by sftim** [comment]
/retitle [WIP] Document KubeletEnsureSecretPulledImages feature

so that Prow knows it's a draft

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -584,6 +585,12 @@ Each feature gate is designed for enabling/disabling a specific feature:
   the `cgroupDriver` configuration setting.
   See [Configuring a cgroup driver](/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver)
   for more details.
+- `KubeletEnsureSecretPulledImages`: add support in kubelet for the `pullIfNotPresent`
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:588]:
```suggestion
- `KubeletEnsureSecretPulledImages`: change the behavior of the `IfNotPresent`
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -584,6 +585,12 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  image pull policy, for ensuring images pulled with pod `imagePullSecrets` are re-authenticated
+  for other pods that do not have the same `imagePullSecret`/auths used to successfully pull
+  the images in the first place.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:591]:
```suggestion
  for other pods that do not have the same `imagePullSecret` authentication information that was used
  to retrieve that Pod's container image(s) in the first place.
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 4]
```diff
@@ -584,6 +585,12 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  This policy change will have no affect on the pull always image pull policy or for images
+  that are preloaded.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:593]:
```suggestion
  Enabling this feature gate has no effect on Pods that use the `Always` image pull policy, nor for the
  `Never` image pull policy or for container images that are preloaded onto the target node.
```

**Review by sftim — COMMENTED** [review]
Hi. Some early feedback - I know this is a work in progress.

**Comment by pacoxu** [comment]
@sftim thanks for your comments. However, the PR for k/k is not merged yet. This will be postponed to v1.30.

I will open a new PR for v1.30 later. I will update it according to your comments in the new PR. Thanks.

**Pull request closed without merging**